### ISSUE TITLE
Comment: injected filetype is used on linewise motions

### ISSUE THREAD

**Issue by przepompownia** [neovim/neovim#31003]
### Problem

See the repro below

### Steps to reproduce

```lua
vim.cmd('echo 1')
return vim.cmd('echo 1')
```
In the above lua file type `gg0fe` then `gcG`. The result is
```lua
"vim.cmd('echo 1')
"return vim.cmd('echo 1')
```

### Expected behavior

```lua
-- vim.cmd('echo 1')
-- return vim.cmd('echo 1')
```
like by `VGgc`.

### Nvim version (nvim -v)

v0.11.0-dev-1067+gb4599acbf

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Debian Sid

### Terminal name/version

alacritty

### $TERM environment variable

alacritty

### Installation

from repo

**Comment by clason** [comment]
This is intended; you're commenting inside a Vimscript context, so that's the commentstring you get. I understand why this would not be desirable, but special-casing this situation would be (way) too complicated.

**Comment by lewis6991** [comment]
I don't think special casing this would be too bad. We just need to make sure the comment context is determined from the position the first comment char is inserted.

We should at least consider a PR if someone wants to fix this.

**Comment by clason** [comment]
Yes, I was too hasty there; the main issue here is the inconsistency of the motion with basic `gcc`, which uses the correct (Lua) commentstring in this situation. 

(Basically, "linewise motion" is actually _not_ linewise; this should be fixed.)
